src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -0,0 +1,874 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon.Options;
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Triggers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Globalization;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics.Tracing;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class CollectionRuleOptionsTests
+    {
+        private const string DefaultRuleName = "TestRule";
+        private const string UnknownEgressName = "UnknownEgress";
+
+        private readonly ITestOutputHelper _outputHelper;
+
+        public CollectionRuleOptionsTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_MinimumOptions()
+        {
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger();
+                },
+                ruleOptions =>
+                {
+                    Assert.Empty(ruleOptions.Filters);
+
+                    ruleOptions.VerifyStartupTrigger();
+
+                    Assert.Empty(ruleOptions.Actions);
+
+                    Assert.Null(ruleOptions.Limits);
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_NoTrigger()
+        {
+            return ValidateFailure(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .AddExecuteAction("echo");
```

Comment:
Sounds like we'll want a UseShellExecute settings on the Execute Action.

Reply:
I would prefer not to allow UseShellExecute in order to be able to capture stdout and stderr. I'll update the test examples to use cmd.exe instead of just calling "echo".